src/rosdiscover/recover/symbolic.py

```diff
@@ -349,6 +349,45 @@ def eval(self, context: SymbolicContext) -> None:
             statement.eval(context)
 
 
+@attr.s(frozen=True, auto_attribs=True, slots=True)
+class SymbolicIf(SymbolicStatement):
+    """Represents a sequence of symbolic if."""
+    true_body: SymbolicCompound
+    false_body: SymbolicCompound
+    condition: SymbolicValue
+
+    def to_dict(self) -> t.Dict[str, t.Any]:
+        return {
+            "kind": "while",
+            "trueBranchBody": [self.true_body.to_dict()],
+            "falseBranchBody": [self.false_body.to_dict()],
+            "condition": [self.condition.to_dict()],
+        }
+
+    def eval(self, context: SymbolicContext) -> None:
+        self.condition.eval(context)
+        self.true_body.eval(context)
+        self.false_body.eval(context)
+
+
+@attr.s(frozen=True, auto_attribs=True, slots=True)
+class SymbolicWhile(SymbolicStatement):
+    """Represents a sequence of symbolic while."""
+    body: SymbolicCompound
+    condition: SymbolicValue
+
+    def to_dict(self) -> t.Dict[str, t.Any]:
+        return {
+            "kind": "while",
+            "body": [self.body.to_dict()],
+            "condition": [self.condition.to_dict()],
+        }
+
+    def eval(self, context: SymbolicContext) -> None:
+        self.condition.eval(context)
```

Comment:
This doesn't look quite right? Maybe we need to chat more about this one? Intuitively, I would have expected an actual while loop here.

Reply:
I've changed it to an if for now. If the condition evaluates to true we would have an endless loop here, which sounds like a bad idea. I think for now we only need the statements inside the loop to be evaluated ones if the condition is satisfiable since this is what happened before as well. Once we actually want them to be evaluated multiple times we can think about a sound way to evaluate this